prusti-viper/src/encoder/mir/pure/specifications/encoder.rs

```diff
@@ -1,824 +1,373 @@
-                                            .map(|x| self.encode_assertion(x))
-                                            .collect::<Result<Vec<vir_high::Expression>, _>>()?
-                                            .into_iter()
-                                            .conjoin(),
-                                    ),
-                                ),
-                            );
-
-                            vec![pre_conjunct, post_conjunct].into_iter().conjoin()
-                        }
-                        _ => unreachable!(),
-                    }
-                } else {
-                    // TODO
-                    true.into()
-                }
-            }
-        })
-    }
-
-    /// Encode a universal or existential quantifer. Encodes type bounds of
-    /// quantified variables as:
-    /// * premises in a universal quantifier, or
-    /// * conjuncts in an existential quantifier.
-    fn encode_quantifier(
-        &self,
-        vars: &typed::QuantifierVars<'tcx>,
-        trigger_set: &typed::TriggerSet,
-        body: &typed::Assertion<'tcx>,
-        exists: bool,
-    ) -> SpannedEncodingResult<vir_high::Expression> {
-        let mut encoded_args = vec![];
-        // FIXME: the type bounds should be part of the type invariant.
-        let mut bounds = vec![];
-        for (arg, ty) in &vars.vars {
-            // TODO: how to get a span for the variable here?
-            //if !self.encoder.is_quantifiable(ty).unwrap() {
-            //    return Err(EncodingError::incorrect(
-            //        "This type cannot be used in quantifiers.",
-            //    ).with_span(?));
-            //}
-
-            // FIXME: figure out from where to get the span and replace “unwrap” with ?
-            let encoded_arg = self
-                .encode_quantifier_arg(*arg, ty, &format!("{}", vars.id))
-                .unwrap();
-            if config::check_overflows() {
-                debug_assert!(self.encoder.env().type_is_copy(ty, ty::ParamEnv::empty()));
-                // FIXME: the type bounds should be part of the type invariant.
-                if let Some((lower_bound, upper_bound)) = self.encoder.get_integer_type_bounds(ty) {
-                    bounds.push(vir_high::Expression::less_equals(
-                        lower_bound,
-                        encoded_arg.clone().into(),
-                    ));
-                    bounds.push(vir_high::Expression::less_equals(
-                        encoded_arg.clone().into(),
-                        upper_bound,
-                    ));
-                }
-            } else if config::encode_unsigned_num_constraint() {
-                if let ty::TyKind::Uint(_) = ty.kind() {
-                    let expr =
-                        vir_high::Expression::less_equals(0u8.into(), encoded_arg.clone().into());
-                    bounds.push(expr);
-                }
+// ================== high ==================
```

Comment:
Could you please split this into separate files because the comments are hard to notice?